### PR TITLE
Add new resource 'kubectl_kustomize_documents'

### PR DESCRIPTION
**Pull request by matyat**
This will allow the use of https://kustomize.io/ to generate a series of
k8s YAML docs, then apply them to a cluster in much the same way as
'kubectl_kustomize_documents'

### CODE REVIEW

**Comment by Logibox** [comment]
As per https://github.com/gavinbunney/terraform-provider-kubectl/issues/78, it's a bit rough and ready. Test cases probably needs expanding but it's a start.

File: go.mod [hunk 1]
```diff
@@ -16,13 +16,14 @@ require (
...
+	k8s.io/cli-runtime v0.21.1
+	k8s.io/client-go v0.21.1
+	k8s.io/kube-aggregator v0.21.1
+	k8s.io/kubectl v0.21.1
+	sigs.k8s.io/kustomize/api v0.8.10
 	sigs.k8s.io/yaml v1.2.0
 )
 
-//replace github.com/Azure/go-autorest v10.15.4+incompatible => github.com/Azure/go-autorest v13.0.0+incompatible
+replace github.com/tencentcloud/tencentcloud-sdk-go v3.0.82+incompatible => github.com/tencentcloud/tencentcloud-sdk-go v1.0.169
```
Comment (Logibox) [inline go.mod:29]: This was required for me to be able to build at all, seems like AFAIK this project has deleted all its tags at some point in the past.

**Comment by schollii** [comment]
Looks good, any chance for a review and merge to master?

**Comment by NicklasWallgren** [comment]
Great addition! @matyat Have you had time to rebase the PR?

**Comment by matyat** [comment]
Hey guys, sorry for the very long delay on this - rebased now. Let me know if any else need adapting, as I realise it's been a while since I opened this PR and the codebase may have moved on a bit since then.

**Comment by burningalchemist** [comment]
Hey folks! Do you have any blockers to review/merge this PR?

**Comment by ritsuki1227** [comment]
@gavinbunney Hi, thank you for this useful module! I'd appreciate it if you review this feature.

**Comment by bbigras** [comment]
Any progress on this?

**Comment by burningalchemist** [comment]
Keep summoning @gavinbunney to the thread. 😃

**Comment by kholisrag** [comment]
any blocker on this one?, we would like this PR to be release soon. Thank You!

**Comment by iuliancristea** [comment]
@gavinbunney this would be immensely useful if merged

**Comment by fdzuJ** [comment]
please @gavinbunney

**Comment by falmar** [comment]
hello @gavinbunney hope this find you well, any plans for this to merged soon?

**Comment by gavinbunney** [comment]
@matyat I know it's been a while since you opened this 😅 It's on my radar to take a deeper look and get it merged in. Appreciate the PR and tests 👍

**Comment by gavinbunney** [comment]
There's a few changes needed to support recent updates to main; will merge this as is and fix it up in there. Thank you for the feature @matyat 👏

**Pull request merged**